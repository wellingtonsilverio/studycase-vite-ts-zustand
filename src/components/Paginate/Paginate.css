.Paginate {
  display: flex;
  align-items: center;
  justify-content: center;

  width: fit-content;

  margin: 0 auto;

  border-radius: 0.4em;

  background: #1a1a1a;
}
.Paginate input {
  height: 24px;

  text-align: center;

  border: none;

  background: #1a1a1a;
}
