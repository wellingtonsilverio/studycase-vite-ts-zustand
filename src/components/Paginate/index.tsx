interface Props {
  amount: number;
  onChange: () => void;
}

const Paginate: React.FC<Props> = ({ amount, onChange }) => {
  return (
    <div>
      <span>{`<`}</span>
      <input defaultValue={amount} />
      <span>{`>`}</span>
    </div>
  );
};

export default Paginate;
