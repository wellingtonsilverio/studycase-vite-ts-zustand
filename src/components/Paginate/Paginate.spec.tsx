import { fireEvent, render, screen } from '@testing-library/react';
import Paginate from '.';

describe('Paginate Component', () => {
  it('should render component', () => {
    const { container } = render(<Paginate />);

    expect(container).toMatchSnapshot();
  });

  it('should show amount pages', () => {
    const amount = 10;
    render(<Paginate amount={amount} />);

    expect(screen.findByText(`${amount}`)).toBeInTheDocument();
  });

  it('should show button to back page', () => {
    const amount = 2;
    render(<Paginate amount={amount} />);

    expect(screen.findByText(`<`)).toBeInTheDocument();
  });

  it('should show ellipsis foward button', () => {
    const amount = 7;
    render(<Paginate amount={amount} />);

    expect(screen.findByText(`>`)).toBeInTheDocument();
  });

  it('should change page after change input', () => {
    const amount = 7;
    render(<Paginate amount={amount} />);

    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: '7' } });

    expect(screen.getByRole('textbox').innerText).toBe('7');
  });

  it('should call onChange after change', () => {
    const amount = 7;
    const fakeFunction = jest.fn();
    render(<Paginate amount={amount} onChange={fakeFunction} />);

    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: '7' } });

    expect(fakeFunction).toHaveBeenCalled();
  });
});
