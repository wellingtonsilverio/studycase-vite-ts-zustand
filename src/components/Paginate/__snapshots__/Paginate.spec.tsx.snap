// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginate Component should render component 1`] = `
<div>
  <div>
    <span>
      &lt;
    </span>
    <input
      value=""
    />
    <span>
      &gt;
    </span>
  </div>
</div>
`;
